File: Tests.Common/Objects/CSCompiler/Lambda.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Linq.Expressions;
using System.Text;
using System.Threading.Tasks;
using static ExpressionToString.Tests.Categories;
using static ExpressionToString.Tests.Functions;

namespace ExpressionToString.Tests.Objects {
    partial class CSCompiler {
        [Category(Lambdas)]
        public static readonly Expression NoParametersVoidReturn = Expr(() => Console.WriteLine());
        
        [Category(Lambdas)]
        public static readonly Expression OneParameterVoidReturn = Expr((string s) => Console.WriteLine(s));
        
        [Category(Lambdas)]
        public static readonly Expression TwoParametersVoidReturn = Expr((string s1, string s2) => Console.WriteLine(s1 + s2));

        [Category(Lambdas)]
        public static readonly Expression NoParametersNonVoidReturn = Expr(() => "abcd");

        [Category(Lambdas)]
        public static readonly Expression OneParameterNonVoidReturn = Expr((string s) => s);

        [Category(Lambdas)]
        public static readonly Expression TwoParametersNonVoidReturn = Expr((string s1, string s2) => s1 + s2);
    }
}

File: Tests.Common.VB/Objects/Lambda.vb
Imports ExpressionToString.Tests.Functions

Namespace Objects
    Partial Public Module VBCompiler
        <Category(Lambdas)>
        Public NoParametersVoidReturn As Expression = Expr(Sub() Console.WriteLine())

        <Category(Lambdas)>
        Public OneParameterVoidReturn As Expression = Expr(Sub(s As String) Console.WriteLine(s))

        <Category(Lambdas)>
        Public TwoParametersVoidReturn As Expression = Expr(Sub(s1 As String, s2 As String) Console.WriteLine(s1 + s2))

        <Category(Lambdas)>
        Public NoParametersNonVoidReturn As Expression = Expr(Function() "abcd")

        <Category(Lambdas)>
        Public OneParameterNonVoidReturn As Expression = Expr(Function(s As String) s)

        <Category(Lambdas)>
        Public TwoParametersNonVoidReturn As Expression = Expr(Function(s1 As String, s2 As String) s1 + s2)
    End Module
End Namespace